src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java

```diff
@@ -433,12 +462,26 @@ public void writeTo(DownloadInstance di, Class<?> clazz, Type type, Annotation[]
                             }
                         }
 
+                        // Default to reading the whole file. We'll count down as we write.
+                        long leftToRead = fileSize;
+                        if (!ranges.isEmpty()) {
+                            // Read a range of bytes instead of the whole file.
+                            // For now we only support a single range.
```

Comment:
In the current implementation, if multiple ranges are supplied, are we simply using the first range, and ignoring the remaining ones? 
I'm wondering if it would be more appropriate to throw an exception ("multiple ranges not supported") instead.

Reply:
OK, I see that you meant to do just that - throw an exception, if more than one range. Except I don't think the code is throwing it now, because of a missing ! in line 651 (?).

Reply:
OK, I deleted one previous comment... It looks like the code IS throwing an exception when multiple ranges are supplied...